lib/ngx/re.lua

```diff
@@ -250,8 +254,23 @@ function _M.split(subj, regex, opts, ctx, max, res)
 
     -- trailing nil for non-cleared res tables
 
-    res[res_idx + 1] = sub(subj, sub_idx)
-    res[res_idx + 2] = nil
+    -- delete empty trailing ones (without max)
+    local remaining_str = sub(subj, sub_idx)
+
+    if max <= 0 and remaining_str == "" then
+        for ety_idx = res_idx, 1, -1 do
+            if res[ety_idx] == "" then
+                res[ety_idx] = nil
+            else
+                res_idx = ety_idx
```

Comment:
There is no need for this assignment here, we could use `[ety_idx + 1] = nil` directly on the below line. Or, we could to `res_idx = ety_idx`, and group the trailing nil assignment below this `if/else` construct and grouping it for both branches, right before the `return`.